src/lib.rs

```diff
@@ -87,12 +87,13 @@ fn process_store(
 fn read(mut cx: FunctionContext) -> JsResult<JsPromise> {
     let rt = runtime(&mut cx)?;
     let channel = cx.channel();
-    let (deferred, promise) = cx.promise();
 
     let asset = cx
         .argument::<JsObject>(0)
         .and_then(|obj| parse_asset(&mut cx, obj))?;
 
+    let (deferred, promise) = cx.promise();
```

Comment:
Not sure if I'm missing anything here, but it looks like the main change here was the order?

In this example, does this mean that if the `let asset` call above errored out, it would panic since it would not run the code in `rt.spawn`, which in turn would not call `deferred.settle_with`, which would lead to the panic?

Reply:
Yes, that's the main change. The change works because the above line `let asset = ...` will return early if the parsing of the argument fails. If it returns early, we never call `deferred.settle_with`, which is what causes the bug.